WcaOnRails/app/models/team_member.rb

```diff
@@ -24,6 +44,34 @@ def cannot_demote_oneself
     end
   end
 
+  validate :delegate_team_must_have_at_least_one_senior_delegate
+  def delegate_team_must_have_at_least_one_senior_delegate
+    if self.team.committee.slug == "wca-delegates-committee"
+      unless self.committee_position.slug == 'senior-delegate' && self.current_member?
```

Comment:
`unless` is harder to read and could be missleading when used with logical AND. (If this seems to be only my personal feeling then feel free to ignore this =) ).